### PR TITLE
hobbes: unstable-2020-05-19 -> unstable-2021-04-28 with LLVM 11 support

### PR DESCRIPTION
**Pull request by hrhino**
###### Motivation for this change

The version of Hobbes currently in nixpkgs uses LLVM 6; the upstream has recently added LLVM 12 support. This switches to that.

Also, the tests now pass (with a slight tweak), and the github URL has been updated (the old one still works, but it redirects here).

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124490` at 65e1dce6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>hobbes</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/development/tools/hobbes/default.nix:4:3:
```
  |
4 |   name = "hobbes";
  |   ^
```

Near pkgs/development/tools/hobbes/default.nix:5:3:
```
  |
5 |   version = "unstable-2021-04-28";
  |   ^
```
</li>
  </ul>
</details>

**Comment by hrhino** [comment]
... should I change it to `pname` too?

File: pkgs/development/tools/hobbes/default.nix [hunk 1]
```diff
@@ -1,29 +1,30 @@
-{ lib, stdenv, fetchFromGitHub, cmake, llvm_6, ncurses, readline, zlib, libxml2 }:
+{ lib, stdenv, fetchFromGitHub, cmake, llvm_12, ncurses, python27, readline, zlib, libxml2 }:
```
Comment (hrhino) [inline pkgs/development/tools/hobbes/default.nix:1]: so, `python27` is needed for the tests, but it seems that it becomes a dependency... does this mean that a _user_ is going to have to have python? or do we skip checks completely for releases?

**Comment by hrhino** [comment]
Well, I'll take @thmzlt's :+1: as a request that I do it; and done.

**Comment by ethancedwards8** [comment]
Result of `nixpkgs-review pr 124490` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hobbes</li>
  </ul>
</details>

File: pkgs/development/tools/hobbes/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/hobbes/default.nix:1]: It is only for tests but we no longer allow new usages of python27. Either get it to work with python3 or disable the tests.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/hobbes/default.nix (hunk 1 above)
Comment (hrhino) [reply pkgs/development/tools/hobbes/default.nix:1]: ack; will do

**Comment by hrhino** [comment]
ty ty

**Pull request merged**